### DISCUSSION TITLE
Lidar in OIGE with skrl

### DISCUSSION THREAD

**Discussion by swimpark** [Toni-SM/skrl#104] [Q&A] [answered]
Hi toni  !

Thanks to your efforts, we are using skrl to easily build a reinforcement learning environment and use it for research.

I am currently facing a problem using lidar in skrl(OIGE).

When I train it with "headless=false", it returns the proper value of depth.

![image](https://github.com/Toni-SM/skrl/assets/54010965/0daf55f5-487c-4b70-bddb-b59fd495adce)

However,  when "headless=True", the wrong value seem to be returned. 

![image](https://github.com/Toni-SM/skrl/assets/54010965/16549523-9a51-4851-aa71-c5b7e759a4e6)

I modified it by referring to the post below, but in my case, the value is fixed to the maximum measurement 
distance.

https://forums.developer.nvidia.com/t/unable-to-get-depth-data-with-lidar-sensor-in-headless-mode/223705

![image](https://github.com/Toni-SM/skrl/assets/54010965/ae04e66e-0b75-44e9-a757-dba4cc1380d6)

From my guess, it seems that the ray beam penetrates the obstacle and cannot be measured.

What steps can be taken in this case?

Thank you in advance !

**Comment by psh9002** [comment, marked as answer]
Hi,swimpark

I answered to your question. Please refer to [your question in Isaac Sim forum](https://forums.developer.nvidia.com/t/lidar-in-oige-with-skrl/262668/7?u=psh9002).

Thanks.
